rs/sns/governance/src/governance.rs

```diff
@@ -696,13 +700,142 @@ pub struct Governance {
     /// The number of proposals after the last time "garbage collection" was run.
     pub latest_gc_num_proposals: usize,
 
+    /// Global lock for all periodic tasks that relate to upgrades - this guarantees
+    /// that they don't interleave with one another.
+    /// `None` means that the lock is not currently held by any task.
+    /// `Some(x)` means that a task is has been holding the lock since timestamp `x`.
+    upgrade_periodic_task_lock: UpgradeLock,
+
     /// Whether test features are enabled.
     /// Test features should not be exposed in production. But, code that should
     /// not run in production can be gated behind a check for this flag as an
     /// extra layer of protection.
     pub test_features_enabled: bool,
 }
 
+/// An upgrade task instance is characterized by its kind (either `CheckUpgradeStatus` or
+/// `RefreshCachedUpgradeSteps`) and an ID.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+enum UpgradeTaskInstance {
+    CheckUpgradeStatus(u64),
+    RefreshCachedUpgradeSteps(u64),
+}
```

Comment:
I'm not sure that this is the right data model, let's think about it more but I think the task ID should probably be stored in the lock itself rather than in UpgradeTaskInstance